Kauri/3box-comments-tutorial.md

```diff
@@ -0,0 +1,333 @@
+    const profile = await Box.getProfile(address);
+
+    // Update the box in our state once we've synced.
+    box.onSyncDone(() => {
+      this.setState({
+        box: box
+      });
+    });
+
+    // Update our state immediately with current values.
+    this.setState({
+      box: box,
+      profile: profile,
+      address: address,
+      ready: true,
+    });
+  }
+
+  ...
+}
+
+...
+```
+
+### Creating the Component
+
+We're finally ready to create our component. We're going to update our [`render()`](https://reactjs.org/docs/react-component.html#render) function so that it renders the component we imported from `3box-comments-react`. Once we're finished with this step, we'll have have a working comment section!
+
+The Comments component takes a few required parameters. These parameters are:
+
+| Property          | Type     | Description                                                                                                                        |
+|-------------------|----------|------------------------------------------------------------------------------------------------------------------------------------|
+| `spaceName`       | `string` | Some [3Box Space](https://docs.3box.io/api/storage) name for your application. You'll usually want to use a single `spaceName` throughout your entire application. |
+| `threadName`      | `string` | Some [3Box Thread](https://docs.3box.io/api/messaging) name for this specific component. Each component typically has its own `threadName`.                          |
+| `adminEthAddr`    | `string` | An Ethereum address for the admin of your application. Admins are able to moderate comments in the comment thread.                 |
+| `box`             | `Object` | A `3box` instance. In our case, the one we initialized and added to our `state`.                                                   |
+| `currentUserAddr` | `string` | The address of the user viewing the component. We pulled this from the `ethereum` instance earlier.
+
+Let's go ahead and create the component using the values we initialized in `login()`. Make sure to replace `spaceName`, `threadName`, and `adminEthAddr` with your own values:
+
+```js
+// src/App.js
+...
+
+class App extends React.Component {
+  ...
+
+  render() {
+    const {
+      box,
+      address,
+      profile,
+      ready,
+    } = this.state;
+
+    return (
+      <div className="App">
+        <div className="comments-container">
+          <BoxComments
+            spaceName="your-3box-space-name"
+            threadName="your-3box-thread-name"
+            adminEthAddr="your-eth-address"
+
+            box={box}
+            currentUserAddr={address}
+          />
+        </div>
+      </div>
+    );
+  }
+}
+
+...
+```
+
+Wait for the development server to recompile the app. You should be able to see the working demo! When you initially load the page, you should get a few popups from your browser wallet. Specifically, you should see:
```

Comment:
@kfichter At this point I get:

```
./node_modules/ipfs-bitswap/src/types/message/index.js
Module not found: Can't resolve 'multicodec/src/name-table' in '/Users/chrisward/Workspace/3box-comments-demo/node_modules/ipfs-bitswap/src/types/message'
```

Are we missing something or is this a 3box bug?

Reply:
Hmm... that seems like a 3box bug. I'll run through the tutorial again tonight to double-check this.

Reply:
Ok just ran through the demo and got the same error. I did some digging and it does seem to be an [issue with the 3box SDK](https://github.com/3box/3box-js/issues/687). Any suggestions on how to handle this?

Reply:
This seems to be a recent issue and one they are looking at. Let's hold for a day or two and see if it's resolved, and if not, try and find another solution.